### PR TITLE
Map PostgreSQL inet to Trino IPADDRESS

### LINKED ISSUE #7540: Trino schema not including all columns from PostgreSQL.
We  are testing a presto instance (on docker)  We connected it to o PostgreSQL database that contains time-series data.
The following columns show on the PostgreSQL database:
image

These are the columns that show on trino-cli. 
image

As you can see the IP column does not show up. (The error that shows up in "stats" is: "line 1:8: Column 'ip' cannot be resolved"

~$ cat power.properties \
connector.name=postgresql  \
connection-url=jdbc:postgresql:/server:port/power \
connection-user=user  \
connection-password=password

( I noticed a similar issue in prestodb: https://github.com/prestodb/presto/issues/10918 )

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Map PostgreSQL inet to Trino IPADDRESS

## Related issues, pull requests, and links

* Fixes #7540

## Documentation

( ) No documentation is needed.
( ) Sufficient documentation is included in this PR.
(x) Documentation PR is available with #5131.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

( ) No release notes entries required.
(x) Release notes entries required with the following suggested text:

```markdown
# PostgreSQL
* Map PostgreSQL `inet` type to Trino `ipaddress` type. ({issue}`7540`)
```

### CODE REVIEW

File: plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java [hunk 1]
```diff
@@ -1253,4 +1267,43 @@ private ColumnMapping uuidColumnMapping()
...
+                        bytes = new byte[16];
+                        bytes[10] = (byte) 0xff;
+                        bytes[11] = (byte) 0xff;
+                        arraycopy(address, 0, bytes, 12, 4);
+                    }
+                    else if (address.length == 16) {
+                        bytes = address;
+                    }
+                    else {
+                        throw new TrinoException(GENERIC_INTERNAL_ERROR, "Invalid InetAddress length: " + address.length);
```
Comment (leveyja) [inline plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java:1290]: Would it be better for trouble shooting to do this:
```
String inetString = resultSet.getString(columnIndex); // save String on the stack for later exception message...
byte[] address = InetAddresses.forString(inetString).getAddress();
....
    throw new TrinoException(GENERIC_INTERNAL_ERROR, format("Invalid InetAddress [%s] - length: %d", inetString, address.length));
```
?

**Review by leveyja — COMMENTED** [review]
Just some general questions + suggestions - I'll leave it to others to approve as I'm not qualified (yet!)

File: plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java [hunk 2]
```diff
@@ -1253,4 +1267,43 @@ private ColumnMapping uuidColumnMapping()
                 (resultSet, columnIndex) -> javaUuidToTrinoUuid((UUID) resultSet.getObject(columnIndex)),
                 uuidWriteFunction());
     }
+
+    private ColumnMapping inetColumnMapping()
+    {
+        return ColumnMapping.sliceMapping(
+                ipAddressType,
+                (resultSet, columnIndex) -> {
```
[the remark this answers is not in the document]
Comment (ebyhr) [reply plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java:1275]: No need to extract a method as the read method isn't reused.

**Comment by ebyhr** [comment]
Closing as PostgreSQL accepts subnet mask that is unsupported in Trino.

**Comment by hashhar** [comment]
> Closing as PostgreSQL accepts subnet mask that is unsupported in Trino.

We can still add a mapping to allow reading and writing values as supported by Trino IPADDRESS type IMO.

**Comment by wendigo** [comment]
@ebyhr maybe we should introduce a new type to express Postgres inet type?

**Comment by ebyhr** [comment]
@hashhar @leveyja Yes, we still can, but I would like to postpone the decision because changing the type mapping (not adding a new mapping) is basically breaking changes.  

@wendigo I guess so.

**Pull request closed without merging**